src/component-library/NumberInput/NumberInput.tsx

```diff
@@ -1,39 +1,56 @@
+import { useNumberField } from '@react-aria/numberfield';
+import { mergeProps } from '@react-aria/utils';
+import type { NumberFieldStateProps } from '@react-stately/numberfield';
+import { useNumberFieldState } from '@react-stately/numberfield';
 import * as React from 'react';
 
-import { InputProps } from '../Input';
+import { chain } from '@/component-library/utils/chain';
+import { useDOMRef } from '@/component-library/utils/dom';
+
 import { BaseNumberInput } from './NumberInput.style';
 
-// `onWheel` prop can't be used with `preventDefault` because
-// React implements passive event listeners.
-const disableChangeOnWheel = (event: MouseEvent) => {
-  event.preventDefault();
+// Prevents the user from changing the input value using mouse wheel
+const handleWheel = (e: WheelEvent) => e.preventDefault();
+
+// Format options for react-stately
+const formatOptions: Intl.NumberFormatOptions = { style: 'decimal', maximumFractionDigits: 20 };
+
+// Static locale for react-stately
+// TODO: To be replaced when we manage our locales
+const locale = 'en-US';
+
+type Props = {
+  onChange?: React.ChangeEventHandler<HTMLInputElement>;
```

Comment:
Added this `onChange`, because the one inherited from `NumberFieldStateProps` is customised: `(value: string) => void`.

We need it to be a native `onChange` so we can easily integrate with `reack-hook-form` `register.`